File: poc-kronos-frontend/src/model/atestado/Atestado.js
class Atestado {
  constructor(dados) {
    this.id = dados ? dados.id : null;
    this.dataEntrega = dados ? dados.dataEntrega : null;
    this.idPessoa = dados ? dados.idPessoa : null;
    this.idTipoFalta = dados ? dados.idTipoFalta : null;
    this.nomeTipoFalta = dados ? dados.nomeTipoFalta : null;
    this.idArqAnexo = dados ? dados.idArqAnexo : null;
    this.dataInicioVigencia = dados ? dados.dataInicioVigencia : null;
    this.dataFinalVigencia = dados ? dados.dataFinalVigencia : null;
    this.idFuncionario = dados ? dados.idFuncionario : null;
    this.idInstituicao = dados ? dados.idInstituicao : null;
    this.numeroDias = dados ? dados.numeroDias : null;
  }

  static validations(required, dataFimMenor) {
    return {
      dataEntrega: { required },
      idPessoa: { required },
      idTipoFalta: { required },
      dataInicioVigencia: { required },
      dataFinalVigencia: {
        required,
        dataFimMenor: dataFimMenor("dataInicioVigencia")
      }
    };
  }

  static validationMessages() {
    return {
      dataEntrega: {
        required: "Data Entrega é obrigatória"
      },
      idPessoa: {
        required: "Pessoa é obrigatória"
      },
      idTipoFalta: {
        required: "Tipo Falta é obrigatória"
      },
      dataInicioVigencia: {
        required: "Data Inicio Vigência é obrigatória"
      },
      dataFinalVigencia: {
        required: "Data Final Vigência é obrigatório",
        dataFimMenor: "Data Final Vigência não pode ser menor que a Data Inicio Vigência"
      }
    };
  }
}
export default Atestado;

File: src/main/java/br/com/kronos/backend/aplicacao/atestado/Atestado.java

package br.com.kronos.backend.aplicacao.atestado;

import static br.com.kronos.backend.aplicacao.validador.Validacoes.exigirNaoNulo;

import java.time.LocalDate;
import java.time.temporal.ChronoUnit;

import lombok.Builder;
import lombok.Getter;

@Getter
public class Atestado {
    
    private Long id;
    private LocalDate dataEntrega; 
    private Long idPessoa;
    private Long idInstituicao;
    private Integer idTipoFalta;
    private Integer idArqAnexo;
    private LocalDate dataInicioVigencia; 
    private LocalDate dataFinalVigencia; 
    private Long idFuncionario;
    private Integer idTipoAtestado;
    private Long numeroDias;
    private Long idContrato;

	@Builder
    public Atestado(Long id, LocalDate dataEntrega, Long idPessoa, Long idInstituicao, Integer idTipoFalta, Integer idArqAnexo, LocalDate dataInicioVigencia,
    		LocalDate dataFinalVigencia, Long idFuncionario, Integer idTipoAtestado, Long numeroDias, Long idContrato) {

		super();
        this.id = id;
        this.dataEntrega = exigirNaoNulo(dataEntrega, "Data Entrega"); 
        this.idPessoa = exigirNaoNulo(idPessoa, "Pessoa");
        this.idInstituicao = exigirNaoNulo(idInstituicao, "Instituição"); 
        this.idTipoFalta = exigirNaoNulo(idTipoFalta, "Tipo de Falta");
        this.idArqAnexo = idArqAnexo == null || idArqAnexo == 0 ? null : idArqAnexo;
        this.dataInicioVigencia = exigirNaoNulo(dataInicioVigencia, "Data Inicio Vigência"); 
        this.dataFinalVigencia = exigirNaoNulo(dataFinalVigencia, "Data Final Vigência"); 
        this.idFuncionario = idFuncionario; 
        this.idTipoAtestado = idTipoAtestado != null ? idTipoAtestado : EnumTipoAtestado.ATESTADO_MEDICO.id(); 
        this.numeroDias = numeroDias != null ? numeroDias : ChronoUnit.DAYS.between(dataInicioVigencia, dataFinalVigencia);
        this.idContrato = idContrato;
      
	}
}